# HG changeset patch
# User Aleksej Saushev <asau@inbox.ru>
# Date 1248119733 14400
#      Mon Jul 20 15:55:33 2009 -0400
# Node ID 160c564d5d25347d6f8fac52d5a20e99b0470c93
# Parent  357cff83985db5175236cded3f978a35a9993c99
initialize floating point values properly for NetBSD systems

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,12 @@
+2009-07-20  John W. Eaton  <jwe@octave.org>
+
+	* interpreter/contributors.in: Add Aleksej Saushev.
+
 2009-06-24  Rafael Laboissiere  <rafael@debian.org>
 
 	* Makefile.in (dist): Drop conf.texi as prerequisite.
 
 2009-06-16  Jaroslav Hajek  <highegg@gmail.com>
 
 	* interpreter/tips.txi: Update.
 
diff --git a/doc/interpreter/contributors.in b/doc/interpreter/contributors.in
--- a/doc/interpreter/contributors.in
+++ b/doc/interpreter/contributors.in
@@ -178,16 +178,17 @@ Andrew Ross
 Mark van Rossum
 Kevin Ruland
 Ryan Rusaw
 Olli Saarela
 Toni Saarela
 Juhani Saastamoinen
 Radek Salac
 Ben Sapp
+Aleksej Saushev
 Alois Schloegl
 Michel D. Schmid
 Julian Schnidder
 Nicol N. Schraudolph
 Sebastian Schubert
 Ludwig Schwardt
 Thomas L. Scofield
 Daniel J. Sebald
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2009-07-20  John W. Eaton  <jwe@octave.org>
+
+	* lo-ieee.cc (octave_ieee_init) [__NetBSD__]: Call nan to
+	initialize Octave_NaN and nanf, to initialize Octave_Float_NaN.
+
 2009-07-11  John W. Eaton  <jwe@octave.org>
 
 	* file-ops.cc (file_ops::symlink, file_ops::readlink):
 	Don't use incorrectly sized OCTAVE_LOCAL_BUFFER.
 
 2009-07-08  John W. Eaton  <jwe@octave.org>
 
 	* dim-vector.h (dim_vector::dim_vector_rep::dim_vector_rep
diff --git a/liboctave/lo-ieee.cc b/liboctave/lo-ieee.cc
--- a/liboctave/lo-ieee.cc
+++ b/liboctave/lo-ieee.cc
@@ -88,16 +88,18 @@ octave_ieee_init (void)
 	      break;
 	    tmp = tmp_inf;
 	  }
 #endif
 
 #if defined (__alpha__) && defined (__osf__)
 	extern unsigned int DQNAN[2];
 	octave_NaN = (*(X_CAST(double *, DQNAN)));
+#elif defined (__NetBSD__)
+	octave_NaN = nan ("");
 #else
 	octave_NaN = tmp_inf / tmp_inf;
         // try to ensure that lo_ieee_sign gives false for a NaN.
         if (lo_ieee_signbit (octave_NaN))
           octave_NaN = -octave_NaN;
 
 #endif
 
@@ -134,17 +136,21 @@ octave_ieee_init (void)
 	  {
 	    float_tmp_inf *= 1e+10;
 	    if (float_tmp_inf == float_tmp)
 	      break;
 	    float_tmp = float_tmp_inf;
 	  }
 #endif
 
+#if defined (__NetBSD__)
+	octave_Float_NaN = nanf ("");
+#else
 	octave_Float_NaN = float_tmp_inf / float_tmp_inf;
+#endif
 	octave_Float_Inf = float_tmp_inf;
 
 	lo_ieee_float tf;
 	tf.word = LO_IEEE_NA_FLOAT;
 	octave_Float_NA = tf.value;
       }
       break;
 
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-07-20  Aleksej Saushev  <asau@inbox.ru>
+
+	* sysdep.cc: Also define BSD_init if __NetBSD__ is defined.
+	(sysdep_init): Also call BSD_init if __NetBSD__ is defined.
+
 2009-07-20  Jaroslav Hajek  <highegg@gmail.com>
 
 	* bitfcns.cc (DO_UBITSHIFT): Avoid overflow.
 	(DO_SBITSHIFT): Fix mask calculation. 
 
 2009-07-17  Benjamin Lindner <lindnerb@users.sourceforge.net>
 
 	* DLD-FUNCTIONS/__magick_read__.cc (F__magick_read__):
diff --git a/src/sysdep.cc b/src/sysdep.cc
--- a/src/sysdep.cc
+++ b/src/sysdep.cc
@@ -90,17 +90,17 @@ along with Octave; see the file COPYING.
 #include "toplev.h"
 #include "utils.h"
 #include "file-stat.h"
 
 #ifndef STDIN_FILENO
 #define STDIN_FILENO 1
 #endif
 
-#if defined (__386BSD__) || defined (__FreeBSD__)
+#if defined (__386BSD__) || defined (__FreeBSD__) || defined (__NetBSD__)
 static void
 BSD_init (void)
 {
 #if defined (HAVE_FLOATINGPOINT_H)
   // Disable trapping on common exceptions.
 #ifndef FP_X_DNML
 #define FP_X_DNML 0
 #endif
@@ -295,17 +295,17 @@ SCO_init (void)
   fpsetmask (~(FP_X_OFL|FP_X_INV|FP_X_DZ|FP_X_DNML|FP_X_UFL|FP_X_IMP));
 #endif
 }
 #endif
 
 void
 sysdep_init (void)
 {
-#if defined (__386BSD__) || defined (__FreeBSD__)
+#if defined (__386BSD__) || defined (__FreeBSD__) || defined(__NetBSD__)
   BSD_init ();
 #elif defined (__MINGW32__)
   MINGW_init ();
 #elif defined (_MSC_VER)
   MSVC_init ();
 #elif defined (NeXT)
   NeXT_init ();
 #elif defined (__EMX__)
